### ISSUE TITLE
Use ResizeObserver in ButtonGroup

### ISSUE THREAD

**Issue by LFDanLu** [adobe/react-spectrum#1115]
# 🙋 Feature Request

ButtonGroup currently uses the window `resize` event to figure out whether or not it should switch from horizontal to vertical orientation when there isn't enough room to render all of its buttons (https://github.com/adobe/react-spectrum/blame/main/packages/%40react-spectrum/buttongroup/src/ButtonGroup.tsx#L69-L79). Now that `useResizeObserver` exists (https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/utils/src/useResizeObserver.ts), we should use it in ButtonGroup and replace the window `resize` listeners.

## 🧢 Your Company/Team

RSP

**Comment by snowystinger** [comment]
one thing to note, whomever works on this, is that inline-flex may cause an issue because the container for the button group won't try to fill the available space like a block level element

**Comment by samiur98** [comment]
Hi @LFDanLu I would like to work and help with this particular issue.

**Comment by samiur98** [comment]
Hello Guys I have implemented this feature and created a pull request. https://github.com/adobe/react-spectrum/pull/1138 
As @snowystinger mentioned inline-flex did cause an issue as the button group did not want to go back from vertical to horizontal when there was available space. 
I was able to fix this issue however by setting the CSS flex-grow property to 1. 
This basically instructs the container to fill space whenever available which I believe is the intended functionality for ButtonGroup
